script/screenshots.js

```diff
@@ -0,0 +1,142 @@
+// This script assumes the following:
+//   1. You've installed wd with `npm install wd'.
+//   2. You've set the environment variables $SAUCE_USERNAME and $SAUCE_ACCESS_KEY.
+//   3. If the environment variable $CIRCLE_ARTIFACTS is not set imanges will be saved in /tmp
+//
+// The intention of this script is that it will be ran from CircleCI
+//
+// Example usage:
+// node screenshots.js http://localhost:9292/test/visual.html
+// node screenshots.js http://google.com
+
+var wd = require('wd');
+var fs = require('fs');
+var username = process.env['SAUCE_USERNAME'];
+var accessKey = process.env['SAUCE_ACCESS_KEY'];
+var baseDir = process.env['CIRCLE_ARTIFACTS'] || '/tmp';
+var url = process.argv[2];
+var dir = baseDir+'/imgs'
+fs.mkdirSync(dir)
+
+var browserVersions = [
+  {
+    'browserName': 'Internet Explorer',
+    'platform': 'Windows XP'
+  },
+  {
+    'browserName': 'Internet Explorer',
+    'platform': 'Windows 7'
+  },
+  {
+    'browserName': 'MicrosoftEdge',
+    'platform': 'Windows 10'
+  },
+  {
+    'browserName': 'Firefox',
+    'platform': 'OS X 10.11'
+  },
+  {
+    'browserName': 'Safari',
+    'platform': 'OS X 10.11'
+  },
+  {
+    'browserName': 'Chrome',
+    'platform': 'OS X 10.11'
+  },
+  {
+    'browserName': 'Firefox',
+    'platform': 'Linux'
+  },
+];
+
+
+browserVersions.forEach(function(cfg) {
+  var browserDriver = wd.remote('ondemand.saucelabs.com', 80, username, accessKey);
+  // The following is in the style of
+  // https://github.com/admc/wd/blob/62f2b0060d36a402de5634477b26a5ed4c051967/examples/async/chrome.js#L25-L40
+  browserDriver.init(cfg, function(err, _, cap) {
+    var browser = cfg.browserName.replace(/\s/g, '_')+(!!cap ? '_'+cap.version : '');
+    var platform = (cap ? cap : cfg).platform.replace(/\s/g, '_');
+    var subDir = dir+'/'+browser+'_'+platform;
+    fs.mkdirSync(subDir)
+
+    if (err) console.log(err);
+
+    browserDriver.get(url, function(err) {
+      if (err) console.log(err);
+      browserDriver.safeExecute('document.documentElement.scrollHeight', function(err,scrollHeight) {
+        if (err) console.log(err);
+        browserDriver.safeExecute('document.documentElement.clientHeight', function(err,viewportHeight) {
+          if (err) console.log(err);
+
+          var position = 0;
```

Comment:
wanna call this `scrollTop`, that's what jQuery calls it, and on anything but the `window` would be the relevant DOM property?